src/components/Message/MessageLivestream.js

```diff
@@ -27,234 +48,320 @@ const optionsSvg =
+                      // @ts-ignore
+                      retryHandler(message);
+                    }
+                  }}
+                >
+                  <svg
+                    width="14"
+                    height="14"
+                    xmlns="http://www.w3.org/2000/svg"
+                  >
+                    <path
+                      d="M7 0a7 7 0 1 0 0 14A7 7 0 0 0 7 0zm.875 10.938a.438.438 0 0 1-.438.437h-.875a.438.438 0 0 1-.437-.438v-.874c0-.242.196-.438.438-.438h.875c.241 0 .437.196.437.438v.874zm0-2.626a.438.438 0 0 1-.438.438h-.875a.438.438 0 0 1-.437-.438v-5.25c0-.241.196-.437.438-.437h.875c.241 0 .437.196.437.438v5.25z"
+                      fill="#EA152F"
+                      fillRule="evenodd"
+                    />
+                  </svg>
+                  {t && t('Message failed. Click to try again.')}
+                </p>
+              )}
+            </div>
 
-  componentWillUnmount() {
-    document.removeEventListener('click', this.hideOptions, false);
-    document.removeEventListener('click', this.hideReactions, false);
-  }
+            {hasAttachment &&
+              Attachment &&
+              attachments.map(
+                /** @type {(item: import('stream-chat').Attachment) => React.ReactElement | null} Typescript syntax */
+                (attachment, index) => (
+                  <Attachment
+                    key={`${message?.id}-${index}`}
+                    attachment={attachment}
+                    actionHandler={propHandleAction || handleAction}
+                  />
+                ),
+              )}
 
-  hasAttachments = () => {
-    const { message } = this.props;
-    return (
-      message && Boolean(message.attachments && message.attachments.length)
-    );
-  };
+            {galleryImages.length !== 0 && <Gallery images={galleryImages} />}
 
-  getAttachments = () => {
-    const { message } = this.props;
-    if (!message || !message.attachments) {
-      return { galleryImages: [], attachments: [] };
+            {message && (
+              <SimpleReactionsList
+                reaction_counts={message.reaction_counts}
+                reactions={message.latest_reactions}
+                handleReaction={propHandleReaction || handleReaction}
+              />
+            )}
+
+            {!initialMessage && message && (
+              <MessageRepliesCountButton
+                onClick={propHandleOpenThread || handleOpenThread}
+                reply_count={message.reply_count}
+              />
+            )}
+          </div>
+        </div>
+      </div>
+    </React.Fragment>
+  );
+};
+
+/**
+ * @type { React.FC<import('types').MessageLivestreamActionProps> }
+ */
+const MessageLivestreamActions = (props) => {
+  const {
+    initialMessage,
+    message,
+    tDateTimeParser: propTDateTimeParser,
+    channelConfig,
+    threadList,
+    handleOpenThread,
+    getMessageActions,
+    messageListRect,
+    isMyMessage,
```

Comment:
missing hooks for some props here, so the component can also be used without the HOC message.